### PR TITLE
fix(expo-notifications): android 11 crash when click foreground notifications

### PR DESCRIPTION
**Pull request by GrinZero**
# Why

This is a long-standing issue, with many recently opened or previously closed issues due to timeout
- https://github.com/expo/expo/issues/29878
- https://github.com/expo/expo/issues/24451
- https://github.com/expo/expo/issues/17531

# How

I have added a version judgment to ensure that versions below Android 12 do not go to the `getNotificationResponseFromBroadcastIntent` function, but instead continue the logic from now on

# Test Plan

By comparing the performance before and after Android 11, and also attempting to test on Android 14, no crashes occurred and normal responses were notified.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-notifications/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,12 @@
 
 - Keep using the legacy event emitter as the module is not fully migrated to Expo Modules API. ([#28946](https://github.com/expo/expo/pull/28946) by [@tsapeta](https://github.com/tsapeta))
 
+## 0.28.10 - 2024-07-05
+
+### 🐛 Bug fixes
+
+- [Android] Android 11 crash when click foreground notifications. ([#30207](https://github.com/expo/expo/pull/30207) by [@GrinZero](https://github.com/GrinZero))
```
Comment (douglowder) [inline packages/expo-notifications/CHANGELOG.md:19]: Your changelog entry should go into the "Unpublished" section; you should not create a new release version section (that is done automatically when we publish a release of this package).

**Review by douglowder — CHANGES_REQUESTED** [review]

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt [hunk 2]
```diff
@@ -454,7 +454,7 @@ open class NotificationsService : BroadcastReceiver() {
       // [notification trampolines](https://developer.android.com/about/versions/12/behavior-changes-12#identify-notification-trampolines)
       // are not allowed. If the notification wants to open foreground app,
       // we should use the dedicated Activity pendingIntent.
-      if (action.opensAppToForeground()) {
+      if (action.opensAppToForeground() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
```
Comment (douglowder) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt:457]: Thanks for this fix!!

File: packages/expo-notifications/CHANGELOG.md (hunk 1 above)
Comment (GrinZero) [reply packages/expo-notifications/CHANGELOG.md]: Okay~The issue has been resolved

**Review by tsapeta — APPROVED** [review]
Thanks for your contribution! 🙏

**Comment by this-is-bgm** [comment]
I feel that my application crashed due to this issue, and I would like to know how long it will take for the expo notification to work properly

**Review by douglowder — APPROVED** [review]
Looks good, once CI finishes, we will merge and get this published.

**Comment by F170x** [comment]
I'm continue having this issue in the latest versions of expo and expo-notifications

```
expo: 53.0.20
expo-notifications: 0.31.4
```

I have errors in Android 11, 12 and 13.

```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3433)
 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3607)
 at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
 at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
 at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2068)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loop(Looper.java:223)
 at android.app.ActivityThread.main(ActivityThread.java:7680)
 at java.lang.reflect.Method.invokeNative(Method.java)
 at java.lang.reflect.Method.invoke(Method.java:423)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
  ```

**Pull request merged**